galsim/wfirst/wfirst_wcs.py

```diff
@@ -0,0 +1,571 @@
+    towards the Sun in the absence of a roll offset, +Y_obs makes a right-handed system.
+
+    Payload coordinate system: +X_pl points along -Y_obs, +Y_pl points along +Z_obs, +Z_pl points
+    along -X_obs (back towards observer).
+
+    Wide field imager (WFI) focal plane assembly (FPA) coordinate system: This is defined by a
+    left-handed system f1, f2, that is rotated by an angle `theta_fpa` with respect to the payload
+    axes.  +f1 points along the long axis of the focal plane, transverse to the radius from the
+    telescope optic axis.  +f2 points radially out from the telescope optic axis, along the narrow
+    dimension of the focal plane.  If +f2 points North, then +f1 points East.  `theta_fpa` is a
+    positive CCW rotation of the f2 axis relative to -Y_pl, and of f1 relative to +X_pl.  In terms
+    of focal plane geometry, if +Y_fp is pointing North, then SCAs 3 and 12 will be at highest
+    declination, 8 and 17 at the lowest.  +Y_fp is aligned with the short axis of the focal plane
+    array.
+
+    There is also a detector coordinate system (P1, P2).  +P1 and +P2 point along the fast- and
+    slow-scan directions of the pixel readout, respectively.
+
+    So, for reference, if the boresight is pointed at RA=90, DEC=0 on March 21st (Sun at vernal
+    equinox), then +X_obs points at (RA,DEC)=(90,0), +Y_obs points North, and +Z_obs points at the
+    Sun.  The payload coordinates are +X_pl points South, -Y_pl points East.  Finally, the FPA
+    coordinate system is defined by +f2 being at a position angle 90+theta_fpa east of North.  If
+    the observatory +Y axis is at a position angle `pa_obsy` East of North, then the focal plane
+    (+f2) is at a position angle pa_fpa = pa_obsy + 90 + theta_fpa.
+
+    @param PA        Position angle of the observatory Y axis, unless `PA_is_FPA=True`, in which
+                     case it's the position angle of the FPA.  Must be provided as a galsim.Angle.
+    @param ra        Right ascension of the center of the FPA, as a galsim.Angle.  Must be provided
+                     with `dec`.
+    @param dec       Declination of the center of the FPA, as a galsim.Angle.  Must be provided
+                     with `ra`.
+    @param pos       A galsim.CelestialCoord indicating the center of the FPA, as an alternative to
+                     providing `ra` and `dec`.
+    @param PA_is_FPA If True, then the position angle that was provided was the PA of the focal
+                     plane array, not the observatory. [default: False]
+    @param as_header If True, then instead of returning a list of WCS objects, return a list of
+                     galsim.FitsHeader objects defining the WCS.
+    @returns a list of WCS or FitsHeader objects for each SCA.
+    """
+    # First and foremost, check for presence of software that can read the WCS:
+    if not _check_software():
+        raise ImportError("Cannot find any software package that can read the WFIRST WCS!")
```

Comment:
I don't see the point of this.  Without this guard, if the appropriate software is not present, then this function will raise an exception.  With this guard, you make sure that it raises an exception instead.  Hm.